src/Compilers/CSharp/Portable/FlowAnalysis/AbstractRegionDataFlowPass.cs

```diff
@@ -44,10 +51,21 @@ public override BoundNode VisitLambda(BoundLambda node)
 
         public override BoundNode VisitLocalFunctionStatement(BoundLocalFunctionStatement node)
         {
+            _hasLocalFunction = true;
```

Comment:
>_hasLocalFunction = true; [](start = 12, length = 25)

Shouldn't this be done conditionally based on whether we are inside the region? Do we have tests for the API when local function is declared and referenced only outside of the region?